Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1706,7 +1707,7 @@ namespace Babylon
             JsRuntimeScheduler m_runtimeScheduler;
             uint32_t m_timestamp{0};
 
-            std::vector<std::pair<const std::string, Napi::FunctionReference>> m_eventNamesAndCallbacks{};
+            std::vector<std::pair<std::string, std::shared_ptr<Napi::FunctionReference>>> m_eventNamesAndCallbacks{};
```

Comment:
Also, why remove const?

Reply:
if the string is declared as const, the "=" operator doesn't get defined for the std::pair. This breaks move operations used in remove_if/erase.